language/stdlib/modules/libra_system_2.move

```diff
@@ -166,53 +176,45 @@ module LibraSystem2 {
             if (i >= size) { break };
             configs_changed = configs_changed || update_ith_validator_info_(validators, i);
         };
-        Transaction::assert(configs_changed, 42);
-        emit_reconfiguration_event();
+        if (configs_changed) {
+            set_validator_set(validator_set);
+        };
     }
 
     // Public getters
 
     // Return true if addr is a current validator
-    public fun is_validator(addr: &address): bool acquires ValidatorSet { is_validator_(addr, &borrow_global<ValidatorSet>(0x1D8).validators) }
+    public fun is_validator(addr: address): bool { is_validator_(addr, &get_validator_set().validators) }
 
     // Returns validator info
     // If the address is not a validator, abort
-    public fun get_validator_info(addr: &address): ValidatorInfo acquires ValidatorSet {
-        let validator_set_ref = borrow_global<ValidatorSet>(0x1D8);
-        let validator_index_vec = get_validator_index_(&validator_set_ref.validators, addr);
+    public fun get_validator_info(addr: address): ValidatorInfo {
+        let validator_set = get_validator_set();
+        let validator_index_vec = get_validator_index_(&validator_set.validators, addr);
         Transaction::assert(!Vector::is_empty(&validator_index_vec), 19);
 
-        *Vector::borrow(&validator_set_ref.validators, *Vector::borrow(&validator_index_vec, 0))
+        *Vector::borrow(&validator_set.validators, *Vector::borrow(&validator_index_vec, 0))
     }
 
     // Return the size of the current validator set
-    public fun validator_set_size(): u64 acquires ValidatorSet {
-        Vector::length(&borrow_global<ValidatorSet>(0x1D8).validators)
+    public fun validator_set_size(): u64 {
+        Vector::length(&get_validator_set().validators)
+    }
+
+    // This function is used in transaction_fee.move to distribute transaction fees among validators
+    public fun get_ith_validator_address(i: u64): address {
+        let validator_set = get_validator_set();
+        let len = Vector::length(&validator_set.validators);
+        Transaction::assert(i < len, 3);
```

Comment:
what's the purpose of the additional bounds check?

Reply:
I was not sure if it is safe to call `Vector::borrow(&validator_set.validators, i)` if the index `i` is out of range... do you know?

Reply:
The trick to these type of questions is to shine the Sam signal

@sblackshear, educate us!

Reply:
`borrow` aborts if the index is out of bounds.

I was about to point to the `Vector` [documentation](https://github.com/libra/libra/blob/master/language/stdlib/modules/vector.move#L11) (which is thorough for the most part), but see that this note is missing for `borrow`...  will  fix!